### PR TITLE
Refactor README, variables, versions

### PR DESCRIPTION
**Pull request by jakemulley**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,12 +1,13 @@
 terraform {
+  required_version = ">= 1.2.5"
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.27.0"
+      version = ">= 4.0.0"
```
Comment (jaskaransarkaria) [inline versions.tf:6]: Is this downgrade intentional?

**Review by jaskaransarkaria — APPROVED** [review]

File: examples/versions.tf [hunk 2]
```diff
@@ -1,12 +1,13 @@
-
 terraform {
   required_version = ">= 1.2.5"
   required_providers {
     aws = {
-      source = "hashicorp/aws"
+      source  = "hashicorp/aws"
+      version = ">= 4.0.0"
```
Comment (jaskaransarkaria) [inline examples/versions.tf:6]: Below we used to require `>= 4.27.0`, is there a reason for `>=4.0.0` over that version?

File: versions.tf (hunk 1 above)
Comment (jakemulley) [reply versions.tf:6]: 👋🏼 Yes, it is. Historically we used to use a generic version in our modules, which happened to be `4.27.0` when `cloud-platform-environments` namespaces' `versions.tf` pointed to `4.27.0` (i.e. it matched). Going forward, Cloud Platform provided modules will only need to bump the _major_ version and we can use the namespaces' `versions.tf` to pin a provider version.

It gets rid of some toil (i.e. updating 4.27.0 to the next minor release) that we haven't always kept on top of.

File: examples/versions.tf (hunk 2 above)
Comment (jakemulley) [reply examples/versions.tf:6]: As discussed in https://github.com/ministryofjustice/cloud-platform-terraform-sqs/pull/51#discussion_r1285877561.

**Pull request merged**